From d7aedb7f733c8beafad04a0d4a2a05da4514f1c3 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Tue, 2 Jan 2018 13:15:49 -0500
Subject: [PATCH 151/153] prueth: skip node table insert for prp emac mode

In PRP EMAC mode, the node table is not applicable and the
code for insert entry to node table can be skipped. Fix this in
the receive packet processing code.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 drivers/net/ethernet/ti/prueth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ti/prueth.c b/drivers/net/ethernet/ti/prueth.c
index 5060881..d0d9794 100644
--- a/drivers/net/ethernet/ti/prueth.c
+++ b/drivers/net/ethernet/ti/prueth.c
@@ -2332,7 +2332,7 @@ static int emac_rx_packet(struct prueth_emac *emac, u16 *bd_rd_ptr,
 	}
 
 	if (PRUETH_HAS_RED(prueth) && !pkt_info.lookup_success) {
-		if (PRUETH_HAS_PRP(prueth)) {
+		if (PRUETH_HAS_PRP(prueth) && !emac->prp_emac_mode) {
 			memcpy(macid,
 			       ((pkt_info.sv_frame) ? nt_dst_addr + 20 :
 				nt_dst_addr + 6),
-- 
1.9.1

